### ISSUE TITLE
Inference.py error because of truncating audio tensors?

### ISSUE THREAD

**Issue by asusdisciple** [Plachtaa/seed-vc#59]
So I think there is a mistake or difference in the behaviour of inference.py ? You seem to truncate the source audio to
30 seconds only in this line:

```
source_audio = source_audio[:sr * 30]
source_audio = torch.tensor(source_audio).unsqueeze(0).float().to(device)
```

So everything beyond 30s is not processed if I am correct. Also in the next line for the reference audio you subtract the length of the source audio, which can result in strange behaviour of ref_audio[:-30000] or files that have a length of 0. Is this intended or did I misunderstand something here?

```
ref_audio = ref_audio[:(sr * 30 - source_audio.size(-1))]
ref_audio = torch.tensor(ref_audio).unsqueeze(0).float().to(device)
```

**Comment by Plachtaa** [comment]
30s is the context window during training so truncation is made to avoid performance degradation caused by long sequences
There is going to be a fix for inference.py

**Comment by asusdisciple** [comment]
yeah that explains it. i thought of inference.py as a way to convert files of arbitrary length.
